app/src/main/java/ch/epfl/sweng/rps/db/FirebaseHelper.kt

```diff
@@ -18,5 +19,87 @@ sealed class FirebaseHelper {
                 uid = uid,
             )
         }
+
+        suspend fun getStatsData (selectMode: Int): MutableList<List<String>> {
+            val userGameList = FirebaseRepository().gamesOfUser(FirebaseRepository().getCurrentUid())
+            val statsResult: MutableList<String> = ArrayList()
+            val allStatsResult: MutableList<List<String>> = java.util.ArrayList()
+            for(userGame in userGameList){
+                val game = FirebaseRepository().getGame(userGame.id)
+                val opponents = game?.players
+                val gameRounds = game?.rounds
+                val allRoundScores = gameRounds?.map {it.value.computeScores() }
+                val userScore= allRoundScores?.asSequence()?.map { scores ->
+                    val max = scores.maxOf { it.points }
+                    if (scores.any {it.points == max &&  it.uid == FirebaseRepository().getCurrentUid() && !scores.all { it.points ==max }})
+                        1
+                    else
+                        0
+
+                }?.sum()
+
+                val roundMode = game?.mode?.rounds
+                //by default 1v1 here
+                val opponentScore = roundMode?.minus(userScore!!)
+                // should be shown like "3 -2 "
+                val score = "$userScore - $opponentScore"
+                val date = SimpleDateFormat("yyyy-MM-dd").format(game?.timestamp?.toDate())
+                    statsResult.add(game!!.id)
+                    statsResult.add(date)
+                    statsResult.add(opponents.toString())
+                    statsResult.add(roundMode.toString())
+                    statsResult.add(score)
+                //mode filter, selectMode = 0 means by default to fetch all result
+                // todo: map selectMode (best of X) in UI to Game round number
+                if(roundMode == selectMode || selectMode == 0) {
+                    allStatsResult.add(statsResult)
+                }
+            }
+            return allStatsResult
+
+
+        }
```

Comment:
The logic here seems wrong to me, you're adding elements to `statsResult` and every iteration, adding the list `statsResult` to `allStatsResult`. Maybe you wanted to have ` val statsResult: MutableList<String> = ArrayList()` *inside the for loop* ?

Also, I'm not a big fan of returning 2d arrays for no specific reason, maybe yo ushould replace the inner list with a class something like 
```kotlin
UserStat(val gameId: String, /*...*/)
```